app/test/unit/git/branch-test.ts

```diff
@@ -186,74 +187,111 @@ describe('git/branch', () => {
       expect(branch).not.toBeNull()
       expect(await getBranches(repository, ref)).toBeArrayOfSize(1)
 
-      await deleteBranch(repository, branch!, null, false)
+      await deleteLocalBranch(repository, branch.name)
 
       expect(await getBranches(repository, ref)).toBeArrayOfSize(0)
     })
+  })
+
+  describe('deleteRemoteBranch', () => {
+    let mockRemote: Repository
+
+    beforeEach(async () => {
+      const path = await setupFixtureRepository('test-repo')
+      mockRemote = new Repository(path, -1, null, false)
+    })
 
-    it('deletes remote branches', async () => {
+    it('delete a local branches upstream branch', async () => {
       const name = 'test-branch'
-      const branch = await createBranch(repository, name, null)
+      const branch = await createBranch(mockRemote, name, null)
       const localRef = `refs/heads/${name}`
 
       expect(branch).not.toBeNull()
-      expect(await getBranches(repository, localRef)).toBeArrayOfSize(1)
 
-      const fork = await setupLocalForkOfRepository(repository)
+      const mockLocal = await setupLocalForkOfRepository(mockRemote)
 
       const remoteRef = `refs/remotes/origin/${name}`
-      const [remoteBranch] = await getBranches(fork, remoteRef)
+      const [remoteBranch] = await getBranches(mockLocal, remoteRef)
       expect(remoteBranch).not.toBeUndefined()
 
-      await checkoutBranch(fork, null, remoteBranch)
-      await git(['checkout', '-'], fork.path, 'checkoutPrevious')
+      await checkoutBranch(mockLocal, null, remoteBranch)
+      await git(['checkout', '-'], mockLocal.path, 'checkoutPrevious')
 
-      expect(await getBranches(fork, localRef)).toBeArrayOfSize(1)
-      expect(await getBranches(repository, localRef)).toBeArrayOfSize(1)
+      expect(await getBranches(mockLocal, localRef)).toBeArrayOfSize(1)
+      expect(await getBranches(mockRemote, localRef)).toBeArrayOfSize(1)
 
-      const [localBranch] = await getBranches(fork, localRef)
+      const [localBranch] = await getBranches(mockLocal, localRef)
       expect(localBranch).not.toBeUndefined()
+      expect(localBranch.upstreamRemoteName).not.toBeNull()
+      expect(localBranch.upstreamWithoutRemote).not.toBeNull()
+
+      if (
+        localBranch.upstreamRemoteName === null ||
+        localBranch.upstreamWithoutRemote === null
+      ) {
+        return
+      }
```

Comment:
I think this is a leftover and should be removed, right?

Reply:
The setup up for the test should ensure these are not null. But, the linter doesn't know that; those these are here to appease it. I added the expect not nulls to error out before this in the case that something would ever change making these actually be null because then the test would not make sense.

Reply:
Ohhh, ok! In other tests I saw we just used `!` in situations like that:
```ts
      await deleteRemoteBranch(
        mockLocal,
        null,
        localBranch.upstreamRemoteName!,
        localBranch.upstreamWithoutRemote!,
        localBranch.tip.sha
      )
```

I'm happy with both approaches, although if you prefer to keep the `if`, I'd add a comment to explain why it's there (in case someone wonders the same I did 🤣 ).